pages/inventories/keycaps/[id].js

```diff
@@ -0,0 +1,328 @@
+      prevColors.filter((existingColor) => existingColor !== color)
+    );
+  };
+
+  const [notes, setNotes] = useLocalStorageState(`notes-${id}`, {
+    defaultValue: [],
+  });
+  const [newNote, setNewNote] = useState("");
+
+  const handleAddNote = () => {
+    if (newNote.trim() === "") return; //no empty notes please
+    if (newNote.length > 100)
+      return alert("Note must be 100 characters or less.");
+
+    const timestamp = new Date().toLocaleString(); //get date and time for note
+
+    setNotes([...notes, { text: newNote, timestamp }]); //add new note to notes array
+
+    setNewNote(""); //clear input field
+  };
+
+  if (error) return <p>Error loading keycap details.</p>;
+  if (!keycaps) return <p>Loading...</p>;
+
+  const kitsAvailable = keycaps.kits?.flatMap((kit) => kit.price_list) ?? [];
+
+  return (
+    <DetailPageContainer>
+      <CloseButton onClick={() => router.push("/inventories/keycaps")}>
+        ×
+      </CloseButton>
+
+      <HeaderSection>
+        <h1>{keycaps.name}</h1>
+        {keycaps.render_pics?.length > 0 && (
+          <HeaderImage>
+            <Image
+              src={keycaps.render_pics[0]}
+              alt={keycaps.name}
+              layout="fill"
+              objectFit="cover"
+            />
+          </HeaderImage>
+        )}
+      </HeaderSection>
+
+      <BoxContainer>
+        <p>
+          <strong>Manufacturer:</strong> {keycaps.keycapstype}
+        </p>
+        <p>
+          <strong>Profile:</strong> {keycaps.profile}
+        </p>
+        <p>
+          <strong>Designer:</strong> {keycaps.designer}
+        </p>
+        <p>
+          <strong>Geekhack Thread:</strong>{" "}
+          <ExternalLink href={keycaps.link} target="_blank">
+            Visit Geekhack
+          </ExternalLink>
+        </p>
+      </BoxContainer>
+
+      <h3>Your Kits</h3>
+      {kitsAvailable.length > 0 ? (
+        <GridContainer>
+          {kitsAvailable.map((kit) => (
+            <KitCard key={kit.name}>
+              {kit.pic && (
+                <Image
+                  src={kit.pic}
+                  alt={kit.name}
+                  width={100}
+                  height={100}
+                  objectFit="cover"
+                />
+              )}
+              <p>{kit.name}</p>
+            </KitCard>
+          ))}
+        </GridContainer>
+      ) : (
+        <p>No kits available for this keycap set.</p>
+      )}
+
+      <h3>Choose 3 Colors</h3>
+      <StyledInput
+        as="select"
+        onChange={handleColorSelect}
+        value=""
+        maxWidth="400px"
+      >
+        <option value="" disabled>
+          -- Choose up to 3 colors --
+        </option>
+        {colorOptions.map((color) => (
+          <option key={color.name} value={color.name}>
+            {color.name}
+            {color.emoji}
+          </option>
+        ))}
+      </StyledInput>
+
+      <GridContainer flexMode padding="10px" border bgColor="#f9f9f9">
```

Comment:
Here again you are nesting divs inside other divs. That's not good practice. Use ul and li instead.